lib/bolt/outputter.rb

```diff
@@ -2,24 +2,25 @@
 
 module Bolt
   class Outputter
-    def self.for_format(format, color, verbose, trace)
+    def self.for_format(format, color, verbose, trace, spin)
       case format
       when 'human'
-        Bolt::Outputter::Human.new(color, verbose, trace)
+        Bolt::Outputter::Human.new(color, verbose, trace, spin)
       when 'json'
-        Bolt::Outputter::JSON.new(color, verbose, trace)
+        Bolt::Outputter::JSON.new(color, verbose, trace, false)
       when 'rainbow'
-        Bolt::Outputter::Rainbow.new(color, verbose, trace)
+        Bolt::Outputter::Rainbow.new(color, verbose, trace, spin)
       when nil
         raise "Cannot use outputter before parsing."
       end
     end
 
-    def initialize(color, verbose, trace, stream = $stdout)
+    def initialize(color, verbose, trace, spin, stream = $stdout)
       @color = color
       @verbose = verbose
       @trace = trace
       @stream = stream
+      @spin = spin
```

Comment:
Since there's an instance variable set to indicate whether the spinner is on, we could use that to 'clear' the spinner character when printing subsequent messages, which would help with console messages having the spinner at the beginning. For example, `print_message` could be updated like so:

```ruby
      def print_message(message)
        @stream.print("\b") if @spin
        @stream.puts(message)
      end
```

Reply:
I don't follow how that improves the feature or code?

Reply:
It removes the spinner character from the beginning of the line. For example:

![image](https://user-images.githubusercontent.com/30373575/101662460-6726bb00-39fe-11eb-9ab3-d4ffe286060a.png)

With a change similar to this:

Reply:
Are you sure you're on the latest? The spinner itself prints a backspace character that should remove it from any output, and I can't reproduce what you're seeing....

Reply:
I'm pretty sure I was... I've been trying to always fetch when testing changes. It seems to be working now, though. Was it an issue with not printing the backspace for the rainbow outputter?

Reply:
Yeah I think I just hadn't pushed it :sweat_smile: